email_mailbox/src/components/MxLoadingWrapper.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Loading from './Loading';
+import { validateDomainMX } from '../utils/ipc';
+
+const statusType = {
+  RUNNING: 'running-animation',
+  STOP: 'stop-animation',
+  COMPLETE: 'complete-animation'
+};
+
+const errorType = {
+  MX_NOT_MATCH: 1,
+  OTHER: 2
+};
+
+const delay = 1800;
+const delayError = 25;
+const delaySuccess = 35;
+
+class MxLoadingWrapper extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      percent: 0,
+      timeout: 0,
+      failed: false,
+      errorType: 0,
+      animationClass: statusType.RUNNING,
+      minutes: 4
+    };
+  }
+
+  componentDidMount() {
+    this.increasePercent();
+  }
+
+  render() {
+    return (
+      <Loading
+        animationClass={this.state.animationClass}
+        percent={this.state.percent}
+        failed={this.state.failed}
+        errorType={this.state.errorType}
+        restart={this.restart}
+        decreaseStep={this.decreaseStep}
+        minutes={this.state.minutes}
+        domain={this.props.domain}
+      />
+    );
+  }
+
+  increasePercent = () => {
+    const percent = this.state.percent + 1;
+    if (percent === 1 || percent === 33 || percent === 66 || percent === 99) {
```

Comment:
What is `1`, `33`, `66` and `99`?

Reply:
are the percent numbers in which the domain is tried to validate , in total i try to validate the domain 4 times